analyzer/src/test/java/com/novoda/sqlite/impl/SQLFileTest.java

```diff
@@ -0,0 +1,96 @@
+package com.novoda.sqlite.impl;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class SQLFileTest {
+    private final static String STATEMENT_CREATE_TABLE = "CREATE TABLE banana(_id INTEGER);";
+    private final static String STATEMENT_ALTER_TABLE = "ALTER TABLE banana ADD COLUMN colour TEXT;";
+    private final static String NEW_LINE = "\n";
+    private final static String SPACE = " ";
+    private final static String[] EXPECTED_STATEMENTS = new String[]{STATEMENT_CREATE_TABLE, STATEMENT_ALTER_TABLE};
+    private final static String LINE_COMMENT = "-- this is a line comment";
+    private static final String BLOCK_COMMENT = "/* this is a \n block comment \n spanning over \n multiple lines */";
+    private final static String STATEMENT_FOR_COMMENTS_WITHIN_TESTS_1 = "CREATE TABLE";
+    private final static String STATEMENT_FOR_COMMENTS_WITHIN_TESTS_2 = "banana(_id INTEGER);";
+
+    @Test
+    public void givenSQLFile_whenParseAndGetStatements_thenGetCorrectStatements() throws IOException {
```

Comment:
`parse` simply builds the statements, a private field that can only be accessed by `getStatements`.